File: 0. Easy/367. Valid Perfect Square.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        if num == 1:
            return True

        l, r = 1, num // 2

        while l <= r:
            mid: int = (l + r) // 2
            square: int = mid * mid

            if square == num:
                return True

            if square < num:
                l: int = mid + 1
            else:
                r: int = mid - 1

        return False

File: 0. Easy/367. Valid Perfect Square.java
class Solution {
    public boolean isPerfectSquare(int num) {
        for(long i=0; i<=num; i++)
        {
            long square = i*i;
            if(square == num) return true;
            else if (square > num) break;
        }
        return false;
    }
}